File: gc-starter/file-spring-boot-starter/src/main/java/com/gc/starter/file/SmartFileProperties.java
package com.gc.starter.file;

import lombok.Getter;
import lombok.Setter;
import org.springframework.boot.context.properties.ConfigurationProperties;

/**
 * @author shizhongming
 * 2020/1/15 9:03 下午
 */
@ConfigurationProperties(prefix = "gc.file")
@Getter
@Setter
public class SmartFileProperties {

    private String actuatorType = ActuatorTypeEnum.mongoDB.name();

    private String basePath;
}

File: smart-starter/file-spring-boot-starter/src/main/kotlin/com/smart/starter/file/SmartFileProperties.kt
package com.smart.starter.file

import org.springframework.boot.context.properties.ConfigurationProperties

/**
 * 文件配置类
 * @author ming
 * 2019/6/14 下午8:56
 */
@ConfigurationProperties(prefix = "smart.file")
class SmartFileProperties {
    // 执行器类型
    var actuatorType = ActuatorTypeEnum.mongoDB.name

    var basePath: String? = null
}